test/common_test/CMakeLists.txt

```diff
@@ -24,21 +24,23 @@ set(TEST_INSTALL_DIR ${CMAKE_INSTALL_LIBEXECDIR}/gz/${GZ_DESIGNATION}${PROJECT_V
 include(GzPython)
 
 function(configure_common_test PHYSICS_ENGINE_NAME test_name)
-  set(target_name ${test_name}_${PHYSICS_ENGINE_NAME})
-  add_test(NAME ${target_name}
-    COMMAND
-      ${test_name}
-      $<TARGET_FILE:${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin>
-      --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/${target_name}.xml
-  )
+  if(NOT SKIP_${PHYSICS_ENGINE_NAME} AND NOT INTERNAL_SKIP_${PHYSICS_ENGINE_NAME})
```

Comment:
The `INTERNAL_SKIP_*` variables are set when dependencies that are `REQUIRED_BY` a particular component are not found (see [GzFindPackage.cmake](https://github.com/gazebosim/gz-cmake/blob/gz-cmake3/cmake/GzFindPackage.cmake#L285-L296)). It's not documented behavior, so it's brittle to depend on, but I couldn't figure out a better way, other than reproducing all the `REQUIRED_BY` logic within this file, which would be a pain. I initially tried `if(TARGET ${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin)`, but that didn't work because the `test` folder is added by [GzConfigureBuild.cmake](https://github.com/gazebosim/gz-cmake/blob/gz-cmake3/cmake/GzConfigureBuild.cmake#L161-L238) before the component folders are added and the corresponding targets created. Reversing this order in `gz-cmake` is possible, but it may cause other problems, and would require a `gz-cmake` release. Using `INTERNAL_SKIP_*` seems least bad to me, though it's partly moot if we can't apply a similar change to `gz-rendering`